src/DataStreams/finalizeBlock.cpp

```diff
@@ -17,7 +17,10 @@ namespace DB
             {
                 current.type = unfinalized_type->getReturnType();
                 if (current.column)
-                    current.column = typeid_cast<const ColumnAggregateFunction &>(*current.column).convertToValues();
+                {
+                    auto mut_column = (*std::move(current.column)).mutate();
+                    current.column = typeid_cast<ColumnAggregateFunction *>(mut_column.get())->convertToValues();
```

Comment:
You're not checking for null result, so probably `assert_cast` is more suitable.